src/matrix/room/timeline/Timeline.js

```diff
@@ -206,16 +215,22 @@ export class Timeline {
 
     // used in replaceEntries
     static _entryUpdater(existingEntry, entry) {
+        // ensure other entries for which this existingEntry is a context point to the new entry instead of existingEntry
+        existingEntry.contextForEntries?.forEach(event => event.setContextEntry(entry));
         entry.updateFrom(existingEntry);
         return entry;
     }
 
     /** @package */
     replaceEntries(entries) {
         this._addLocalRelationsToNewRemoteEntries(entries);
+        this._updateEntriesNotInTimeline(entries);
+        this._loadContextEntriesWhereNeeded(entries);
```

Comment:
Not sure why we would need to load context after an event has been updated in any way (redacted mostly, and later on edited)? What could update an entry that changes the context entry that it needs? Relations can't be edited AFAIK...